client/src/ExtensionState.ts

```diff
@@ -119,52 +125,84 @@ export class State {
     return State.client.onReady();
   }
 
-
   // creates a server for the given server binary
-  private static async startServerProcess(serverBin: string): Promise<any> {
-    // test whether java and z3 binaries can be used:
-    Helper.log("Checking Java...");
-    const javaPath = await Helper.getJavaPath();
-    await Helper.spawn(javaPath, ["--version"]);
-    Helper.log("Checking Z3...");
-    const z3Path = Helper.getZ3Path(Helper.isNightly());
-    await Helper.spawn(z3Path, ["--version"]);
+  private static async startServerProcess(serverBin: string): Promise<StreamInfo> {
+    const javaPath = await State.checkDependenciesAndGetJavaPath();
+
+    // spawn Gobra Server and get port number on which it is reachable:
+    const portNr = await new Promise((resolve:(port: number) => void, reject) => {
```

Comment:
I assume that you cannot just use a fixed port as some clients might use the port for some obfuse application?

Reply:
You could pass a given port to the server but as you said, there is no guarantee that this one is free. Hence, the IDE client (for non-development purposes) leaves the port unspecified such that the server will instruct the OS to choose an arbitrary port. The consequence however is that we only know the port number after starting to listen on that particular port and thus the server writes the final port number wrapped in a special character sequence to standard output and the client in turn needs to parse that.